package.cmake

```diff
@@ -61,7 +61,7 @@ set(CPACK_DEBIAN_BASE_FILE_NAME "${CPACK_DEBIAN_FILE_NAME}.deb")
 string(REGEX REPLACE "^(${CMAKE_PROJECT_NAME})" "\\1-dev" CPACK_DEBIAN_DEV_FILE_NAME "${CPACK_DEBIAN_BASE_FILE_NAME}")
 
 #deb package tooling will be unable to detect deps for the dev package. llvm is tricky since we don't know what package could have been used; try to figure it out
-set(CPACK_DEBIAN_DEV_PACKAGE_DEPENDS "libboost-all-dev;libssl-dev;libgmp-dev")
+set(CPACK_DEBIAN_DEV_PACKAGE_DEPENDS "libboost-all-dev, libssl-dev, libgmp-dev")
```

Comment:
The documentation suggests it is not a list
https://cmake.org/cmake/help/latest/cpack_gen/deb.html#variable:CPACK_DEBIAN_PACKAGE_DEPENDS
In particular the example doesn't use a list.

afaict prior to cmake 3.16 this gets plumbed through to the package spec verbatim. This is why leap-dev only has an invalid spec on ubuntu 18 (20 & 22 are fine). It's not clear if this change to allow a list in cmake 3.16+ was an intentional change in cmake or not -- I found no documentation on the matter.

Reply:
@spoonincode I think line 70 is correct and the proposed fix should work; however, setting boost, ssl, and gmp as a single list item is less intuitive than setting them as 3 items. (See my suggestion above.)

Reply:
I believe this works as is in CMake v3.16 and later; however, we are targeting a minimum version of 3.8.

Regarding line 70, here a `dpkg-query` example of output:
```bash
$ dpkg-query -S llvm | cut -d: -f1
llvm-11-dev
llvm-11-dev
libclang-common-11-dev
llvm-11-dev
libclang-common-9-dev
libclang-common-11-dev
...
```
I expect that becomes a list something like `llvm-11-dev;llvm-11-dev;libclang-common-11-dev;...`.

Maybe it would be better to leave line 64 and 70 as is and add a list join in the code below? [I think this is what later version of CMake do](https://gitlab.kitware.com/cmake/cmake/-/blob/v3.16.0/Modules/Internal/CPack/CPackDeb.cmake#L474). 

Something like this around line 73?
```cmake
# Convert list to comma separated string for older versions of CMake.
if(${CMAKE_VERSION} VERSION_LESS "3.16.0")
   list(JOIN CPACK_DEBIAN_DEV_PACKAGE_DEPENDS ", " CPACK_DEBIAN_DEV_PACKAGE_DEPENDS)
endif()
```

Reply:
> ```
> set(FOO "a;b;c")
> set(BAR "a" "b" "c")
> ```
> 
> Is equivalent (I agree the latter is more readable though). So `set(CPACK_DEBIAN_DEV_PACKAGE_DEPENDS "libboost-all-dev" "libssl-dev" "libgmp-dev")` will not work on cmake 3.10 because cmake 3.10 does not treat this variable as a list: it just gets plumbed through to the package spec verbatim.
> 
> I did go ahead and try it
> 
> ```
> # git diff
> diff --git a/package.cmake b/package.cmake
> index 3ec67d290..81b698ed5 100644
> --- a/package.cmake
> +++ b/package.cmake
> @@ -61,7 +61,7 @@ set(CPACK_DEBIAN_BASE_FILE_NAME "${CPACK_DEBIAN_FILE_NAME}.deb")
>  string(REGEX REPLACE "^(${CMAKE_PROJECT_NAME})" "\\1-dev" CPACK_DEBIAN_DEV_FILE_NAME "${CPACK_DEBIAN_BASE_FILE_NAME}")
> 
>  #deb package tooling will be unable to detect deps for the dev package. llvm is tricky since we don't know what package could have been used; try to figure it out
> -set(CPACK_DEBIAN_DEV_PACKAGE_DEPENDS "libboost-all-dev;libssl-dev;libgmp-dev")
> +set(CPACK_DEBIAN_DEV_PACKAGE_DEPENDS "libboost-all-dev" "libssl-dev" "libgmp-dev")
>  find_program(DPKG_QUERY "dpkg-query")
>  if(DPKG_QUERY AND OS_RELEASE MATCHES "\n?ID=\"?ubuntu" AND LLVM_CMAKE_DIR)
>     execute_process(COMMAND "${DPKG_QUERY}" -S "${LLVM_CMAKE_DIR}" COMMAND cut -d: -f1 RESULT_VARIABLE LLVM_PKG_FIND_RESULT OUTPUT_VARIABLE LLVM_PKG_FIND_OUTPUT)
> ```
> 
> ```
> # ar x leap-dev_4.0.0-dev-ubuntu18.04_amd64.deb
> # tar xf control.tar.gz
> # cat control
> Package: leap-dev
> Version: 4.0.0-dev
> Section: devel
> Priority: optional
> Architecture: amd64
> Depends: libboost-all-dev;libssl-dev;libgmp-dev;llvm-7-dev, libc6 (>= 2.25), libgcc1 (>= 1:3.0), libssl1.1 (>= 1.1.0), libstdc++6 (>= 7), zlib1g (>= 1:1.1.4)
> Conflicts: eosio, mandel
> Installed-Size: 85454
> Maintainer: EOS Network Foundation
> Description: headers and libraries for native contract unit testing
> ```
> 
> Still has semicolons.

Sorry i forgot to add abit more information about it. but yes, this is the problem i was having with the depends line including semicolon and was generated with cmake 3.10.2